### PR TITLE
Testgen external commands

### PR DESCRIPTION
**Pull request by andrey-kuprianov**
Part 4 of #529. See also #414. Add to Testgen utilities to simplify calling external commands, and also specifically to call the Apalache model checker, and JSON Artifact Translator (jsonatr).

* [X] Referenced an issue explaining the need for the change
* [X] Updated all relevant documentation in docs
* [X] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

**Comment by andrey-kuprianov** [comment]
Please review this one after #547 is merged -- there is a dependency.

**Review by romac — APPROVED** [review]
Looks great, just a couple nitpicks :)

File: testgen/src/apalache.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    let model = env.read_file(&test.model).unwrap();
+    let mut new_model = String::new();
+    for line in model.lines() {
+        if line.starts_with(&inv) {
+            // invariant already present; skip mutation
+            new_model.clear();
+            break;
+        }
+        if line.starts_with("======") {
+            new_model += &(inv.clone() + " == ~" + &test.test + "\n")
```
[the remark this answers is not in the document]
Comment (andrey-kuprianov) [reply testgen/src/apalache.rs]: nice:) thanks!

File: testgen/src/apalache.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            ApalacheRun::Deadlock(_) => "Apalache has found a deadlock; please inspect your model and test",
+            ApalacheRun::ModelError(_) => "Apalache failed to process the model; please check it",
+            ApalacheRun::Unknown(_) => "Apalache has generated an unknown outcome; please contact Apalache developers",
+            ApalacheRun::Timeout(_) => "Apalache failed to generate a counterexample within given time; consider increasing the timeout, or changing your test",
+        }
+    }
+}
+
+pub fn run_apalache_test(dir: &str, test: ApalacheTestCase) -> io::Result<ApalacheRun> {
+    let inv = test.test.clone() + "Inv";
```
[the remark this answers is not in the document]
Comment (andrey-kuprianov) [reply testgen/src/apalache.rs]: done; thanks:)

**Comment by andrey-kuprianov** [comment]
@romac I've addressed your suggestions; nice:) Could you please take a look and merge if everything is ok? Thanks!

**Review by romac — APPROVED** [review]

**Pull request merged**